pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IdleTimer.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.data.manager.realtime;
+
+/**
+ * The IdleTimer is responsible for keeping track of 2 different idle times:
+ *  - The stream idle time which resets every time we remake the stream consumer.
+ *    This depends on the user configured "idle.timeout.millis" stream config.
+ *  - the total idle time which only resets when we consume something.
+ *
+ * This is not a running timer. It only advances as long as we keep calling "markIdle".
```

Comment:
Why not just tracking the start time, and ask for `nowMs` when reading the idle time? If the start time is 0, return 0 then

Reply:
Yes, tracking both start times sounds good. My question is about whether we should calculate idle time at marking time or reading time. It would be more accurate if we calculate idle time at reading time

Reply:
i was mostly following the current convention, but you're right that it makes less sense with a separate class. I've changed the logic to calculate idleness on read rather than on "mark idle" time.